packages/ember-runtime/lib/mixins/enumerable.js

```diff
@@ -849,6 +850,35 @@ export default Mixin.create({
   },
 
   /**
+    Returns a new enumerable that contains only items containing a unique property value.
+    The default implementation returns an array regardless of the receiver type.
+
+    ```javascript
+    var arr = [{ value: 'a' }, { value: 'a' }, { value: 'b' }, { value: 'b' }];
+    arr.uniqBy('value');  // [{ value: 'a' }, { value: 'b' }]
+    ```
+
+    @method uniqBy
+    @return {Ember.Enumerable}
+    @public
+  */
+
+  uniqBy(key) {
```

Comment:
This must also be behind a feature flag. You should be able to `reopen` the mixin:

```js
if (isEnabled('my-feature')) {
  Enumerable.reopen({
    uniqBy() { // ...
    }
  });
}
```

Additionally this requires tests!